script/reproducible/verify.bat

```diff
@@ -0,0 +1,129 @@
+@ECHO OFF
+set CHECK_VERSION=%1
+SHIFT
+
+set buildir=C:\projects\
+
+REM Interpret arguments
+:loop
+IF NOT "%1"=="" (
+    IF "%1"=="-buildir" (
+        SET buildir=%2
+        SHIFT
+    )
+    SHIFT
+    GOTO :loop
+)
+
+@ECHO ON
+
+set ELECTRON_OUT_DIR=Default
+set ELECTRON_ENABLE_STACK_DUMPING=1
+
+REM Check for disk space
+Rem        543210987654321
+Set "Blank=               "
+Set "GB100=   107374182400"
+
+for /f "tokens=2" %%A in (
+  'wmic LogicalDisk Get DeviceID^,FreeSpace ^| find /i "C:"'
+) Do Set "FreeSpace=%Blank%%%A"
+Set "FreeSpace=%FreeSpace:~-15%"
+
+Echo FreeSpace="%FreeSpace%"
+Echo    100 GB="%GB100%"
+
+If "%FreeSpace%" gtr "%GB100%" (
+  Echo yes enough free space
+) else (
+  Echo not enough free space - 100GB 
+  exit 5
+)
+
+REM Install missing dependencies
+powershell -command "& { iwr https://raw.githubusercontent.com/electron/electron/%CHECK_VERSION%/script/setup-win-for-dev.bat -OutFile C:\install.bat }"
+cmd /c C:\install.bat
+
+
+set GN_CONFIG=release
+
+echo "Building $env:GN_CONFIG build"
+SET PATH=%PATH%;C:\depot_tools\
+SET PATH=%PATH%;C:\Python27\
+git config --global core.longpaths true
+if not exist %buildir% mkdir %buildir%
+cd %buildir%
+set CHROMIUM_BUILDTOOLS_PATH=%buildir%\src\buildtools
+set SCCACHE_PATH=%buildir%\src\electron\external_binaries\sccache.exe
+REM set GYP_MSVS_OVERRIDE_PATH=C:\BuildTools\
+set GYP_MSVS_VERSION=2019
+set DEPOT_TOOLS_WIN_TOOLCHAIN=0
+REM set GCLIENT_EXTRA_ARGS="%GCLIENT_EXTRA_ARGS% --custom-var=\"checkout_boto=True\" --custom-var=\"checkout_requests=True\""
+cmd /C gclient config --name "src\electron" --unmanaged %GCLIENT_EXTRA_ARGS% "https://github.com/electron/electron"
+cmd /C gclient sync -f --with_branch_heads --with_tags

```

Comment:
Why is this run with `cmd /c` instead of just directly invoking `gclient`?

Reply:
remember to move to `call` every `cmd /c`. What is happening here is that it is exiting after the gclient sync execution. If you execute a nested .bat script, for legacy reason, when you `exit` also the father `.bat` is signalled to exit.

Reply:
`cmd /C` spawn a new shell with a separated environment, (inherithed), `call` should instead spawn in the caller environment, and can change it accordingly.

Reply:
ah, so it's to ignore failures?

Reply:
not really failures, in `.bat` failures are always ignored unless you check the return value of every call/exe invocation. It's that whenever you invoke another .bat script (not an .exe), the exit signal will bounce back to the root .bat unless you have a `call` or `cmd /C` and will trigger a exit on the main root .bat, in our case `verify.bat`. I was confused too by this behavior initially, but it's due to `bat` being very legacy friendly.

Reply:
This behavior does not have any similarity with how exit and shell nesting is handled on posix shells. Basically you have this weird behaviour or:

1. `call another.bat` that is similar to `. another.sh`
2. `cmd /C another.sh` that is similar to `sh another.sh`

`another.bat` invocation without anything is not equivalent to `./another.sh`. It's more like both `root.bat` and `another.bat` are having a shared `exit` signal handler and when the first bat invoked exit, it invokes the `exit` signal and bounce the exit.